addons/udes_stock/models/stock_quant.py

```diff
@@ -50,6 +52,24 @@ def get_quantities_by_key(self, get_key=lambda q: q.product_id, only_available=F
             products[get_key(quant)] += value
         return products
 
+    def group_quantity_by_product(self, only_available=False):
+        """
+        Returns a dictionary with the total quantity per product,
+        mapped by product_id.
```

Comment:
nit: add that we remove products that do not have any quantity

Reply:
We should also say `mapped by product` instead of `mapped by product_id`, as we use the product record, not ID.

Reply:
Isn't this the same as: get_quantities_by_key ?

Reply:
Yep exactly the same, have removed and replaced.